cloud/aws/providerconfig/v1alpha1/types.go

```diff
@@ -178,17 +178,28 @@ type VPC struct {
 type Subnet struct {
 	ID string `json:"id"`
 
-	VpcID            string `json:"vpcId"`
-	AvailabilityZone string `json:"availabilityZone"`
-	CidrBlock        string `json:"cidrBlock"`
-	IsPublic         bool   `json:"public"`
+	VpcID            string  `json:"vpcId"`
+	AvailabilityZone string  `json:"availabilityZone"`
+	CidrBlock        string  `json:"cidrBlock"`
+	IsPublic         bool    `json:"public"`
+	RouteTableID     *string `json:"routeTableId"`
+	NatGatewayID     *string `json:"natGatewayId"`
 }
 
 // Subnets is a slice of Subnet.
 type Subnets []*Subnet
 
+// ToMap returns a map from id to subnet.
+func (s Subnets) ToMap() (res map[string]*Subnet) {
+	for _, x := range s {
+		res[x.ID] = x
+	}
+
+	return
+}
+
 // FilterPrivate returns a slice containing all subnets marked as private.
-func (s Subnets) FilterPrivate() (res []*Subnet) {
+func (s Subnets) FilterPrivate() (res Subnets) {
```

Comment:
Similar question here as well.

Reply:
That's true for the map, but append() will allocate the array and re-assign it, so this is safe.

Reply:
Good to know, thanks!